python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py

```diff
@@ -0,0 +1,250 @@
+
+from Ganga.GPIDev.Adapters.IPrepareApp import IPrepareApp
+from Ganga.GPIDev.Schema import Schema, Version, SimpleItem, GangaFileItem
+
+from Ganga.GPIDev.Lib.File.File import ShareDir
+from Ganga.Core import ApplicationConfigurationError, GangaException
+
+from Ganga.Utility.logging import getLogger
+
+from Ganga.GPIDev.Base.Proxy import getName
+
+from os import rename, path, makedirs, chdir
+import shutil
+from Ganga.Utility.files import expandfilename
+
+from Ganga.GPIDev.Lib.File.LocalFile import LocalFile
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+
+from GangaDirac.Lib.Backends.DiracBase import DiracBase
+
+import tempfile
+import time
+import subprocess
+
+logger = getLogger()
+
+def _exec_cmd(cmd, cwdir):
```

Comment:
Do we want yet another way to run external commands? Why not `execute` or `Shell`? Also, this function will never time-out an overly-long process which can cause a hang.